compiler/rustc_parse/src/parser/item.rs

```diff
@@ -1700,23 +1708,33 @@ impl<'a> Parser<'a> {
 /// The function decides if, per-parameter `p`, `p` must have a pattern or just a type.
 type ReqName = fn(Edition) -> bool;
 
+/// Parsing configuration for functions.
+/// This include the edition-specific name requirements, plus information on whether the
+/// function is allowed to go without a body.
+#[derive(Clone, Copy)]
+pub(crate) struct FnParseMode {
+    pub req_name: ReqName,
+    pub req_body: bool,
+}
+
```

Comment:
Could you extend the docstring ("what, why and where" this affects the parser), add a docstring to each field so it's more immediately understood what each of them do.

Reply:
Okay, I've added more comments to it.